helix-core/src/main/java/org/apache/helix/HelixAdmin.java

```diff
@@ -420,6 +420,24 @@ void addStateModelDef(String clusterName, String stateModelDef, StateModelDefini
    */
   void removeCloudConfig(String clusterName);
 
+  /**
+   * Get the topology of a specific cluster
+   * @param clusterName
+   */
+  Map<String, List<String>> getAllTopology(String clusterName);
+
+  /**
+   * Get all the instances under the fault zone
+   * @param clusterName
+   */
+  Map<String, List<String>> getInstancesUnderFaultZone(String clusterName);
+
+  /**
+   * Get all the instances whose domain config is not valid
+   * @param clusterName
+   */
+  List<String> getInvalidInstances(String clusterName);
+
```

Comment:
Didn't we agree to add one method that returns ClusterTopology object here?

Reply:
If we return an object, users will still need to parse it by themselves. It's a lot of work if you look at the functions in ClusterTrie. I don't think we want to have this requirement on them. Or you mean users need to call the functions inside ClusterTrie? then they need to know all the available functions? Please let me know.

Reply:
Discussed offline, and we will only keep one API in HelixAdmin. Updated.